src/plugins/usage_collection/server/usage_counters/usage_counters_service.ts

```diff
@@ -152,32 +156,46 @@ export class UsageCountersService {
     );
   }
 
-  private createUsageCounter = (type: string): UsageCounter => {
-    if (this.counterSets.get(type)) {
-      throw new Error(`Usage counter set "${type}" already exists.`);
+  private createUsageCounter = (
+    domainId: string,
+    soType: UsageCounterSavedObjectType = SERVER_COUNTERS_SAVED_OBJECT_TYPE
+  ): IUsageCounter => {
+    if (this.counterSets.get(domainId)) {
+      throw new Error(`Usage counter set "${domainId}" already exists.`);
     }
 
     const counterSet = new UsageCounter({
-      domainId: type,
+      domainId,
       counter$: this.source$,
+      soType,
     });
 
-    this.counterSets.set(type, counterSet);
+    this.counterSets.set(domainId, counterSet);
```

Comment:
`domainId` could clash for N different `soType`s.

Reply:
The `UsageCountersService` should ensure that a certain domainId is not registered more than once (line 163)